### PR TITLE
Provide access to raw content stream

### PR DESCRIPTION
**Pull request by friedelwolff**
This provides a WordList in the raw order (content stream order) of
the document.

### CODE REVIEW

**Comment by izderadicka** [comment]
Could you please add test case for you changes?

**Comment by friedelwolff** [comment]
I extended dump_file.py to be able to dump the file in raw order. I also added a test document that will clearly show the difference between the two approaches.

I realise now I changed the signature for the class Document. Should we rather update it so that it doesn't require an update to call sites?

File: pdfparser/poppler.pyx [hunk 1]
```diff
@@ -104,11 +112,13 @@ cdef class Document:
         PDFDoc *_doc
         int _pg
         PyBool phys_layout
+        PyBool raw_order
         double fixed_pitch
-    def __cinit__(self, char *fname, PyBool phys_layout=False, double fixed_pitch=0):
+    def __cinit__(self, char *fname, PyBool raw_order=False, PyBool phys_layout=False, double fixed_pitch=0):
```
Comment (izderadicka) [inline pdfparser/poppler.pyx]: Could you move raw_order argument to last position - to make it compatible with previous version of API?

File: pdfparser/poppler.pyx [hunk 2]
```diff
@@ -508,10 +529,74 @@ cdef class Line:
...
+                                   )
+                self._fonts.append(last_font)
+            #and then text as UTF-8 bytes
+            s=w.getText()
+            #print s.getCString(), w.getLength(), len(s.getCString())
+            words.append(s.getCString().decode('UTF-8')) # decoded to python unicode string
+            del s
+            # must have same ammount of bboxes and characters in word
+            assert len(words[-1]) == wlen
+        self._text= u''.join(words)
```
Comment (izderadicka) [inline pdfparser/poppler.pyx]: As per this line we get text without spaces - what about to put extra bbox for each new space (calculated from end and start of surrounding words) and also put extra font info for spaces - from last or first character of surrounding word?

Comment (friedelwolff) [inline pdfparser/poppler.pyx]: As far as I know, it depends on the PDF producer whether or not spaces will be in the raw stream. So they might be there already, or not.

My own need is for a stream that contains only the raw data. To calculate whether there should be spaces between consecutive "words" is a non-trivial problem (and part of what I'm doing with this stream). I realise that poppler has an implementation for this, but I need to build my own on top of the raw data. The "words" that the raw stream will give us is not guaranteed to be true words. They can easily be single characters from words, for example. 

What does bother me is that my approach throws away the current "word" information from the stream, so it isn't really a WordList (as the name suggest). Maybe ``.text`` is not the right solution, and I should rather implement a list type. Then I can get access to the raw data through ``__iter__()`` or ``__getitem__()``. How does that sound? I guess ``.text`` could provide a more readable version (as you prefer).

**Comment by izderadicka** [comment]
Hi,

so if I understand correctly,  what you need is really a 'raw stream of 
characters' as they appear on the page -
Than I think some more appropriate name for class would be appropriate 
-   RawChars? RawText?   whatever is sounds more correct.
As in this case text in not reconstructed to any logical structures,  I 
think most logical would be API consistent with Line -  e.g. text, 
char_boxes and char_fonts properties.

I.

**Pull request closed without merging**